samcli/lib/providers/sam_function_provider.py

```diff
@@ -160,20 +160,29 @@ def _convert_sam_function_resource(name, resource_properties, layers, ignore_cod
         samcli.commands.local.lib.provider.Function
             Function configuration
         """
-
         codeuri = SamFunctionProvider.DEFAULT_CODEURI
+        inlinecode = None
         imageuri = None
         packagetype = resource_properties.get("PackageType", ZIP)
         if packagetype == ZIP:
-            codeuri = SamFunctionProvider._extract_sam_function_codeuri(
-                name, resource_properties, "CodeUri", ignore_code_extraction_warnings=ignore_code_extraction_warnings
-            )
-            LOG.debug("Found Serverless function with name='%s' and CodeUri='%s'", name, codeuri)
+            if "InlineCode" in resource_properties and resource_properties["InlineCode"] is not None:
```

Comment:
It would be much simplier to just attempt to get InlineCode. If the dict doesn't have InlineCode, it will automatically be set to `None`. Which reduces the complexity here.